doc/azure_ci.md

```diff
@@ -0,0 +1,160 @@
+<div class="hidden-warning"><a href="https://docs.haskellstack.org/"><img src="https://cdn.jsdelivr.net/gh/commercialhaskell/stack/doc/img/hidden-warning.svg"></a></div>
+
+# Azure CI
+
+This page documents how to use Stack on [Azure
+CI](http://dev.azure.com/). We assume you have basic familiarity with
+Azure Pipelines. We provide two fully baked example files ready to be
+used on your projects:
+
+* [The simple Azure configuration](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/azure/azure-simple.yml)
+  is intended for applications that do not require multiple GHC
+  support or cross-platform support. It builds and tests your project
+  with just the settings present in your `stack.yaml` file.
+* The complex Azure configuration is intended for projects that need
+  to support multiple GHC versions and multiple OSes, such as open
+  source libraries to be released to Hackage. It tests against
+  cabal-install, as well as Stack on Linux and macOS. The
+  configuration is significantly more involved to allow for all of
+  this branching behavior. In the provided template, we use the Linux
+  build configuration to test against various combinations of
+  cabal-install and stack. The macOS and Windows configuration is used
+  only for testing Stack build configuration. These are the files for
+  the complex configuration:
+  - [azure-pipelines.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/azure/azure-pipelines.yml) : This is the starter file used by the Azure CI.
+  - [azure-linux-template.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/azure/azure-linux-template.yml) : Template for Azure Linux build
+  - [azure-osx-template.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/azure/azure-osx-template.yml) : Template for Azure macOS build
+  - [azure-windows-template.yml](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/azure/azure-windows-template.yml) : Template for Azure Windows build
+
+  __NOTE__: It is likely going to be necessary to modify this configuration to
+  match the needs of your project, such as tweaking the build matrix to alter
+  which GHC versions you test against, or to specify GHC-version-specific
+  `stack.yaml` files if necessary. Don't be surprised if it doesn't work the
+  first time around. See the multiple GHC section below for more information.
+
+Each of these configurations is ready to be used immediately. Steps to make them work:
+* For simple Azure configuration, copy-paste the
+  [azure-simple](https://raw.githubusercontent.com/commercialhaskell/stack/stable/doc/azure/azure-simple.yml)
+  file into `azure-pipelines.yml` in the root of your repository.
+* For complex Azure configuration, you need to take the above four
+files and put them in the root of your repository.
```

Comment:
I see in the Stack project itself, we have a `.azure` hidden directory where most of the files go. That seems more elegant than adding a bunch of root level files.

Reply:
The reason I did it like that was because when you create pipeline using the Azure Web interface, it will write down a file named `azure-pipelines.yml` in the root repository. I will modify the documentation to make it into `.azure`.

Reply:
For the complex configuration, I have modified the docs such that except the main driver file - all others are kept inside `.azure`.